src/preprocess.py

```diff
@@ -1,12 +1,17 @@
 import os
 import pickle as pkl
 
+from collections import OrderedDict
```

Comment:
You might Want to change to OrderedDict in the typing of functions that manipulate the collection as well.

Reply:
done everywhere